### PR TITLE
Added support to bind NotificationHubClient

### LINKED ISSUE #86: [NotificationHub]Add support to bind NotificationHubClient

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #86

### CODE REVIEW

**Comment by pragnagopa** [comment]
@brettsam - Please review

File: src/WebJobs.Extensions.NotificationHubs/Config/NotificationHubsConfiguration.cs [hunk 1]
```diff
@@ -47,19 +59,36 @@ public void Initialize(ExtensionConfigContext context)
...
+            string resolvedConnectionString = ResolveConnectionString(attribute.ConnectionStringSetting);
+            string resolvedHubName = ResolveHubName(attribute.HubName);
+            INotificationHubClientService service = GetService(resolvedConnectionString, resolvedHubName);
+
+            return service.GetNotificationHubClient();
+        }
+
+        internal INotificationHubClientService GetService(string connectionString, string hubName)
+        {
+            return ClientCache.GetOrAdd(new Tuple<string, string>(connectionString, hubName), (c) => NotificationHubClientServiceFactory.CreateService(c.Item1, c.Item2));
```
Comment (brettsam) [inline src/WebJobs.Extensions.NotificationHubs/Config/NotificationHubsConfiguration.cs]: >new Tuple<string, string>(connectionString, hubName) [](start = 40, length = 52)

Should the hubname comparison be case sensitive or case insensitive? Not sure how it works. But you should add a test using a hubname of "TEST" and "test" and see if it behaves the way you want.

File: test/WebJobs.Extensions.Tests/Extensions/NotificationHubs/NotificationHubEndToEndTests.cs [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                .Returns(serviceMock.Object);
+
+            var testTrace = new TestTraceWriter(TraceLevel.Warning);
+
+            //Act
+            RunTest("Outputs", factoryMock.Object, testTrace);
+
+            // Assert
+            factoryMock.Verify(f => f.CreateService(ConfigConnStr, DefaultHubName), Times.Once());
+            serviceMock.Verify(m => m.SendNotificationAsync(It.IsAny<Notification>(), It.IsAny<string>()), Times.Exactly(9));
```
Comment (brettsam) [inline test/WebJobs.Extensions.Tests/Extensions/NotificationHubs/NotificationHubEndToEndTests.cs]: >Times.Exactly(9) [](start = 107, length = 16)

For the sake of completion, I'd have your Collectors each add at least one too. Then just bump this up.

File: src/WebJobs.Extensions.NotificationHubs/Config/NotificationHubsConfiguration.cs (hunk 1 above)
Comment (pragnagopa) [inline src/WebJobs.Extensions.NotificationHubs/Config/NotificationHubsConfiguration.cs]: Good point. Verified hubname is case insensitive. Will add tests and update key to be case insensitive as well.

**Comment by pragnagopa** [comment]
@brettsam - SO?

**Comment by brettsam** [comment]
:shipit:

**Pull request closed without merging**